### ISSUE TITLE
Incompatibility with DocPad v6.70

### ISSUE THREAD

**Issue by zenorocha** [hurrymaplelad/docpad-plugin-rss#8]
I just upgraded my DocPad project to v6.70 which supports Node v0.12 and I got this error with your plugin:

```
warning: Failed to load the plugin: docpad-plugin-rss at /myProject/node_modules/docpad-plugin-rss
The error follows:
warning: An error occured:
Error: Use CoffeeScript.register() or require the coffee-script/register module to require .coffee.md files.
  at Object._base.(anonymous function) [as .coffee] (/myProject/node_modules/docpad/node_modules/cson/node_modules/coffee-script/lib/coffee-script/coffee-script.js:210:17)
  at Module.load (module.js:355:32)
  at Function.Module._load (module.js:310:12)
  at Module.require (module.js:365:17)
  at require (module.js:384:17)
  at PluginLoader.load (/myProject/node_modules/docpad/out/lib/plugin-loader.js:131:28)
  at /myProject/node_modules/docpad/out/lib/docpad.js:2563:27
  at PluginLoader.unsupported (/myProject/node_modules/docpad/out/lib/plugin-loader.js:106:7)
  at /myProject/node_modules/docpad/out/lib/docpad.js:2547:25
  at /myProject/node_modules/docpad/out/lib/plugin-loader.js:56:18
  at /myProject/node_modules/docpad/out/lib/plugin-loader.js:90:20
  at evalmachine.<anonymous>:336:14
  at /myProject/node_modules/docpad/node_modules/safefs/node_modules/graceful-fs/graceful-fs.js:102:5
  at FSReqWrap.oncomplete (evalmachine.<anonymous>:99:15)
```

**Comment by zenorocha** [comment]
This seems to be related to an upgrade to [CoffeeScript 1.7.0](http://coffeescript.org/#1.7.0):

> When requiring CoffeeScript files in Node you must now explicitly register the compiler. This can be done with require `coffee-script/register` or `CoffeeScript.register()`. Also for configuration such as Mocha's, use coffee-script/register.

**Comment by balupton** [comment]
@hurrymaplelad the fix here is to publish the compiled version of the plugin.

To do this:

1. Grab the latest base files from https://github.com/bevry/base

1. Update your `package.json` `main` variable to be `"main": "./out/rss.plugin.js",` instead. You may want to take a look 

2. Run `npm install --save-dev projectz` to install projectz as a dev dependency

3. Run `cake prepublish` to verify everything is okay

4. Commit your changes

5. Run `cake publish` to do the release


Seems the issue here is actually in our "Write a Plugin" guide - https://docpad.org/docs/plugin-write - which is pretty outdated. I'll update it now.

**Comment by balupton** [comment]
I've updated the plugin writing guide https://docpad.org/docs/plugin-write and started a discussion on how to make it even better here https://github.com/docpad/docpad/issues/924

**Comment by hurrymaplelad** [comment]
Good catch @zenorocha.  Thanks for the details @balupton. May be a while before I have time to run through this, PR welcome.

**Comment by zenorocha** [comment]
Hmmm, I really want to fix it because this affects an important project but I never created a DocPad plugin before.

**Comment by balupton** [comment]
For what it works, v6.71.0 of DocPad re-adds support for uncompiled plugins, so it will work - however there will be a warning about it.

**Comment by balupton** [comment]
Sending a PR now.

**Comment by balupton** [comment]
PR sent.

**Comment by hurrymaplelad** [comment]
Published as v2.2.0

**Comment by hurrymaplelad** [comment]
Big thanks to @balupton
